site/src/pages/WorkspacesPage/WorkspacesButton.tsx

```diff
@@ -0,0 +1,246 @@
+    .slice(0, 10);
+}
+
+function WorkspaceResultsRow({ template }: { template: Template }) {
+  return (
+    <Link
+      key={template.id}
+      component={RouterLink}
+      // Sending user directly to workspace creation page for UX
+      // reasons; avoids extra clicks on the user's part
+      to={`/templates/${template.name}/workspace`}
+      sx={{
+        outline: "none",
+        "&:focus": {
+          backgroundColor: (theme) => theme.palette.action.focus,
+        },
+        "&:hover": {
+          textDecoration: "none",
+          backgroundColor: (theme) => theme.palette.action.hover,
+        },
+      }}
+    >
+      <Box
+        sx={{
+          display: "flex",
+          columnGap: COLUMN_GAP,
+          alignItems: "center",
+          paddingX: 2,
+          paddingY: 1,
+          overflowY: "hidden",
+        }}
+      >
+        <Avatar
+          src={template.icon}
+          fitImage
+          alt={template.display_name || "Coder template"}
+          sx={{
+            width: `${ICON_SIZE}px`,
+            height: `${ICON_SIZE}px`,
+            fontSize: `${ICON_SIZE * 0.5}px`,
+            fontWeight: 700,
+          }}
+        >
+          {template.display_name || "-"}
+        </Avatar>
+
+        <Box
+          sx={{
+            lineHeight: 1,
+            width: "100%",
+            overflow: "hidden",
+            color: "white",
+          }}
+        >
+          <Box
+            component="p"
+            sx={{
+              marginY: 0,
+              paddingBottom: 0.5,
+              overflow: "hidden",
+              textOverflow: "ellipsis",
+              whiteSpace: "nowrap",
+            }}
+          >
+            {template.display_name || "[Unnamed]"}
+          </Box>
+
+          <Box
+            component="p"
+            sx={{
+              marginY: 0,
+              fontSize: 14,
+              color: (theme) => theme.palette.text.secondary,
+            }}
+          >
+            {/*
+             * There are some templates that have -1 as their user count –
+             * basically functioning like a null value in JS. Can safely just
+             * treat them as if they were 0.
+             */}
+            {template.active_user_count <= 0
+              ? "No"
+              : template.active_user_count}{" "}
+            developer
+            {template.active_user_count === 1 ? "" : "s"}
+          </Box>
+        </Box>
+      </Box>
+    </Link>
+  );
+}
+
+export function WorkspacesButton() {
+  const organizationId = useOrganizationId();
+  const permissions = usePermissions();
+
+  const templatesQuery = useQuery({
+    ...templates(organizationId),
+
+    // Creating icons via the selector to guarantee icons array stays as stable
+    // as possible, and only changes when the query produces new data
+    select: (templates) => {
+      return {
+        list: templates,
+        icons: templates.map((t) => t.icon),
```

Comment:
This is necessary for the image pre-loading, but since the feature was removed at the last minute, it's not relevant right now.

Not sure if it's worth removing any data transformations for the moment, and adding them back once the other PR is ready